src/ImageLoaders/IHex32/IntelHex32Loader.cs

```diff
@@ -0,0 +1,242 @@
+﻿#region License
+/* 
+ * Copyright (C) 2017-2018 Christian Hostelet.
+ * inspired by work of:
+ * Copyright (C) 1999-2017 John Källén.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using Reko.Core;
+using Reko.Core.Services;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Reko.ImageLoaders.IntelHex32
+{
+    /// <summary>
+    /// An Intel Hexadecimal 32-bit object format image (a.k.a. IHEX32) loader.
+    /// </summary>
+    public class IntelHex32Loader : ImageLoader
+    {
+
+        //TODO: As Intel Hex specs do not specify any ordering of records, we should be able, getting a new record,
+        // to add it at tail, at head, or merge with already loaded records - or to create a new memory chunk.
+        // For the time being we assume we are safe and Hex records are contiguous (not overlapping) and somehow sorted in increasing load address order.
+
+        #region Helper classes
+
+        internal class MemChunk
+        {
+            public readonly Address BaseAddress;
+            public List<byte> Datum;
+            public Address EndAddress => BaseAddress + Datum.Count;
+
+            public MemChunk(Address bAddr)
+            {
+                BaseAddress = bAddr;
+                Datum = new List<byte>();
+            }
+
+            public MemChunk(uint address) : this(Address.Ptr32(address))
+            {
+            }
+
+            public bool Contains(Address addr) => (BaseAddress <= addr && EndAddress > addr);
+
+            public bool IsAtTail(Address addr) => addr == EndAddress;
+
+            public bool IsAtHeadOf(Address addr) => EndAddress == addr;
+
+        }
+
+        /// <summary>
+        /// List of memory chunks.
+        /// </summary>
+        internal class MemoryChunksList : IEnumerable<MemChunk>
+        {
+            private Address currAddr = null;
```

Comment:
To be consistent with the bulk of the Reko codebase, prefer `addrCur`, `addrNext` etc. That is: the (abbreviated) type followed by its function/use.